contracts/multisig-prover/src/encoding/bcs.rs

```diff
@@ -0,0 +1,165 @@
+use bcs::to_bytes;
+use cosmwasm_std::HexBinary;
+use itertools::Itertools;
+
+use crate::error::ContractError;
+
+use super::Data;
+
+pub fn command_params(
+    source_chain: String,
+    source_address: String,
+    destination_address: String,
+    payload_hash: HexBinary,
+) -> Result<HexBinary, ContractError> {
+    let ret = to_bytes(&(
+        source_chain,
+        source_address,
+        <[u8; 32]>::try_from(&HexBinary::from_hex(&destination_address)?.to_vec()[..32])
+            .expect("couldn't convert destination_address to 32 byte array"), // TODO: is this right? Why are addresses 32 bytes?
+        payload_hash.to_vec(),
```

Comment:
I wanna ask the same question. Is this right? Why are addresses 30 bytes?

Reply:
I fixed this but forgot to update the comment somehow. They're 32 bytes

Reply:
It's only OK to panic if some other stage of the message processing has already validated it. Otherwise, one can send any invalid-according-to-destination-chain message to us and make us panic.

Reply:
So I thought about this more and I realized we shouldn't panic here. This is the only place we check the destination address. The abi module returns an error here if the destination address is not a valid evm address. So we should do the same in this module. There's an argument to be had to move the check to somewhere earlier in the pipeline. But for now, we should check here and return a result.

For command ids, we should panic, because command ids are not user input and should never be invalid.

Reply:
ya, `command_params` can return error, the `encode` function should panic as we controls all inputs